flare_splunk_integration/appserver/static/javascript/views/setupConfiguration.js

```diff
@@ -56,7 +68,34 @@ async function saveConfiguration(splunkSdk, serverKey, tenantId) {
     }
 }
 
-function savePassword(storage, storageRealm, key, value) {
+function doesPasswordExist(
+  storage,
+  storageRealm,
+  key,
+) {
+  var passwordList = storage.list();
+  const passwordId = storageRealm + ":" + key + ":";
+
+  for (var index = 0; index < passwordList.length; index++) {
+    if (passwordList[index].name === passwordId) {
+      return true
+    }
+  }
+  return false
+};
+
+async function savePassword(storage, storageRealm, key, value) {
+    var passwordExists = doesPasswordExist(storage, storageRealm, key);
+    if (passwordExists) {
+      const passwordId = storageRealm + ":" + key + ":";
+      storage.del(passwordId);
+    }
+
+    while (passwordExists) {
+      storage = await promisify(storage.fetch)();
+      passwordExists = doesPasswordExist(storage, storageRealm, key);
+    }
```

Comment:
A few things here:
1 - Looks to be a mix of 4 and 2 spaces or maybe tab and 2 spaces. 
2 - We should probably add an `.editorconfig` file.
3 - I remember you explaining to me that you can't just update a value in the password storage, but this looks a bit off to me. Why the while loop? If it really does need to work this way, it's probably worth a comment to explain to whoever else comes through here to maintain it.

Reply:
I'm assuming that `storage.delete()` is async so the while loop is us waiting until the delete has completed. Is that correct?